tools/mpy-tool.py

```diff
@@ -843,7 +843,7 @@ def freeze_mpy(base_qstrs, raw_codes):
     print('const char mp_frozen_mpy_names[] = {')
     for rc in raw_codes:
         module_name = rc.source_file.str
-        print('"%s\\0"' % module_name)
+        print('"{frozen}/%s\\0"' % module_name)
```

Comment:
This will increase code size of firmware by an amount proportional to the number of frozen files.  I think it would be better to have a bit of code in `py/builtinimport.c` which knows about the special frozen path.

Reply:
Very good point. I just reviewed my change, and I think, if I make `do_load` and `mp_import_stat_any` test for the `{frozen}/` prefix and strip (skip) it before passing the path into `mp_find_frozen_module` and `mp_frozen_stat` (i.e. the functions already meant for loading frozen files will not get the `{frozen}/` prefix), then all will work, and we don't need the prefix in the frozen filenames. What do you think?

I intend to make a commit that reflects that change. But since the PR is so old, is it ok to rebase it on top of current master and force-push?

Reply:
For now, I have made a proposed patch: https://gist.github.com/wnienhaus/4492a4128d74e19f1248deab7ed31f7c - to show in code what I mean. There's one ugly part in the patch (around the source_name for frozen strings) - see comment in the gist.

But as I was testing my change, a new idea came up: would it perhaps make sense to implement the entire frozen file handing via VFS, mounted to /.frozen/ ? This would potentially allow removing all special frozen file logic from everywhere, and we'd simply have one more directory to search like any other directory? It feels like a bigger change, but maybe it's a cleaner approach? (I do see that VFS is a compile time directive, so I guess some ports don't have VFS, but could it be that ports with frozen file support always have VFS?)

Reply:
@wnienhaus feel free to rebase and force-push to this branch your latest implementation, that's the best approach.  Then we can re-review it in full.

Re VFS approach: this is an interesting idea, but I don't think it'll work without a lot of changes, because frozen code is nothing like a filesystem (eg there's no flat data to read for any given file, instead it's a set of nested data structures).